soc/arm/nuvoton_npcx/npcx7/soc.c

```diff
@@ -46,3 +46,9 @@ NPCX_REG_OFFSET_CHECK(uart_reg, UFRCTL, 0x026);
 /* GPIO register structure check */
 NPCX_REG_SIZE_CHECK(gpio_reg, 0x008);
 NPCX_REG_OFFSET_CHECK(gpio_reg, PLOCK_CTL, 0x007);
+
+/* PWM register structure check */
+NPCX_REG_SIZE_CHECK(pwm_reg, 0x00E);
+NPCX_REG_OFFSET_CHECK(pwm_reg, PWMCTL, 0x004);
+NPCX_REG_OFFSET_CHECK(pwm_reg, DCR, 0x006);
+NPCX_REG_OFFSET_CHECK(pwm_reg, PWMCTLEX, 0x00C);
```

Comment:
lower-case hex

Reply:
Here is our discussion for this in the other PR. https://github.com/zephyrproject-rtos/zephyr/pull/27578#discussion_r474969730. The reason that I use 12-bits offset is that all npcx module register sizes won't exceed 4KB although we reserved 8K for them. If you think it's a crucial code style must follow, I will add one more commit in this PR to fix all of them in soc.c.